src/functional/iterator/embedded.py

```diff
@@ -701,25 +699,57 @@ def _get_axes(
     )
 
 
+def _single_vertical_idx(indices, column_axis_idx, column_index):
+    transformed = tuple(
+        index if i != column_axis_idx else column_index for i, index in enumerate(indices)
+    )
+    return transformed
+
+
 def _make_tuple(
```

Comment:
This function returns different types depending on the input values in a predictable way. If it doesn't become too complex to make it work with mypy, I would try to define overloads to express this behavior:

```python
@overload
def _make_tuple(
    field_or_tuple: LocatedField,
    indices: FieldIndexOrIndices
    *,
    column_axis: Literal[None],
) -> npt.DTypeLike:
    ...

@overload
def _make_tuple(
    field_or_tuple: LocatedField,
    indices: FieldIndexOrIndices,
    *,
    column_axis: Tag,
) -> Column:
    ...

@overload
def _make_tuple(
    field_or_tuple: tuple[LocatedField | Column, ...],
    indices: FieldIndexOrIndices,
    *,
    column_axis: Literal[None],
) -> tuple[LocatedField | Column, ...]:
    ...

@overload
def _make_tuple(
    field_or_tuple: tuple[LocatedField | Column, ...],
    indices: FieldIndexOrIndices,
    *,
    column_axis: Tag,
) -> Column:
    ...


```

Reply:
done